Added information about Hamming distance and extended other notes about Codes

spdsslg opened the pull request.

iwashis commented on chapters/Codes.tex:
```diff
@@ -29,15 +29,33 @@ \section{Encoding and Decoding Functions}
 \noindent \textbf{Encoding function:} A map $\xi: \mathbb{Z}_p^k \to \mathbb{Z}_p^n$ is called an \emph{encoding function} (or \emph{coding function}) if it is one-to-one. In other words, $\xi$ takes each possible message (an element of $\mathbb{Z}_p^k$) and encodes it as a longer string (an element of $\mathbb{Z}_p^n$). The image of $\xi$, denoted 
 \[ \mathcal{C} = \xi(\mathbb{Z}_p^k) \subseteq \mathbb{Z}_p^n, \] 
-is the set of all possible encoded messages and is called the \emph{code}. The elements of $\mathcal{C}$ are called \emph{codewords}.
+is the set of all possible encoded messages and is called the \emph{code}. The elements of $\mathcal{C}$ are called \emph{codewords}.We usually require $\xi$ to be injective (distinct messages have distinct codewords). Often $\xi$ is a linear transformation (i.e. $\xi(\mathbf{u}+\mathbf{v})=\xi(\mathbf{u})+\xi(\mathbf{v})$ and $\xi(a\mathbf{u})=a\,\xi(\mathbf{u})$ for $a\in \mathbb{Z}_p$), in whi ...
```
Let's not put the definition of linear codes here - it will appear in the near future.

iwashis commented on chapters/Codes.tex:
```diff
@@ -103,4 +121,90 @@ \subsection*{Example 3: \textsc{PESEL} Check Digit}
+
+\section{Metric Spaces and Other Distance Metrics}
+
+The pair $(\mathbb{Z}_p^n, d_H)$ is an example of a \emph{metric space}. In general, a \textbf{metric space} is a set $X$ equipped with a distance function $d: X \times X \to \mathbb{R}$ (called a \emph{metric}) satisfying the three properties listed above (non-negativity and identity of indiscernibles, symmetry, and triangle inequality). The metric $d(x,y)$ represents the “distance” between poin ...
```
In general, a \textbf{metric space} is a set $X$ equipped with a distance function ->

A \textbf{metric space} is a tuple $(X,d)$, where $X$ is a set and $d$ is a function

iwashis commented on chapters/Codes.tex:
```diff
@@ -103,4 +121,90 @@ \subsection*{Example 3: \textsc{PESEL} Check Digit}
+
+\section{Metric Balls and Error Correction}
+
+Given a metric space $(X,d)$, a \textbf{metric ball} (or \emph{distance ball}) of radius $r$ centered at a point $x \in X$ is the set 
```
Not a metric ball, simply a ball.

iwashis commented:
Otherwise super cool, thanks! Please smooth out the formalities though.

alekseikud commented:
Also add yourself as a commiter

spdsslg commented:
I have deleted the part about linear codes, added a proper part about Hamming distance, changed the definition of a metric space, changed a metric ball to a ball and modified the part about Metric spaces a bit

iwashis commented:
@spdsslg  Thanks!
